### PR TITLE
Develop upstream sync 240401

### PR DESCRIPTION
**Pull request by mmakevic-amd**

### CODE REVIEW

File: third_party/xla/xla/backends/profiler/gpu/rocm_collector.cc [hunk 1]
```diff
@@ -223,7 +223,7 @@ class PerDeviceCollector {
     stats.occupancy_pct /= device_properties_.maxThreadsPerMultiProcessor;
 
     err = hipOccupancyMaxPotentialBlockSize(
```
Comment (i-chaochen) [inline third_party/xla/xla/backends/profiler/gpu/rocm_collector.cc:225]: could you also upstream this change as well?

Comment (mmakevic-amd) [reply third_party/xla/xla/backends/profiler/gpu/rocm_collector.cc:225]: @zahiqbal mentioned he is already doing some changes in profiler, and that he'll add this to his PR as well

File: third_party/xla/xla/service/gpu/conv_algorithm_picker.cc [hunk 2]
```diff
@@ -71,7 +71,7 @@ limitations under the License.
 #include "tsl/platform/logging.h"
 #include "tsl/platform/numbers.h"
 #include "tsl/platform/statusor.h"
-#include "tsl/util/env_var.h"
+#include "xla/tsl/util/env_var.h"
```
Comment (i-chaochen) [inline third_party/xla/xla/service/gpu/conv_algorithm_picker.cc:74]: I think this `env_var.h` is only used in our side due to https://github.com/ROCm/tensorflow-upstream/pull/2467 and it's merged https://github.com/ROCm/tensorflow-upstream/commit/c3d966e559ffd13f6a473d1b57e606314a456ce8 maybe you can upstream that PR as well? cc @pemeliya

File: third_party/xla/xla/service/gpu/gpu_layout_assignment.cc [hunk 3]
```diff
@@ -55,7 +55,7 @@ limitations under the License.
 #include "tsl/platform/errors.h"
 #include "tsl/platform/statusor.h"
 
-#include "tsl/util/env_var.h"
+#include "xla/tsl/util/env_var.h"
```
Comment (i-chaochen) [inline third_party/xla/xla/service/gpu/gpu_layout_assignment.cc:58]: same. this one is actually an old one we haven't upstream for a long while. https://github.com/ROCm/tensorflow-upstream/commit/dfb7dd0b2f7e6c6bbcd3f1dbc9e299be9554b4a7 could you upstream this as well?

**Comment by i-chaochen** [comment]
retest Ubuntu-GPU-single please
retest Ubuntu-CPU please

File: third_party/xla/xla/backends/profiler/gpu/rocm_collector.cc (hunk 1 above)
Comment (draganmladjenovic) [reply third_party/xla/xla/backends/profiler/gpu/rocm_collector.cc:225]: Probably they don't get an error here due to jax build using gcc. Guess we need to check those changes applied to tensorflow-upstream to be sure.

File: tensorflow/compiler/mlir/quantization/tensorflow/python/BUILD [hunk 4]
```diff
@@ -176,12 +176,8 @@ cc_library(
         "//tensorflow/compiler/mlir/quantization/tensorflow/calibrator:calibration_statistics_proto_cc",
         "//tensorflow/core/protobuf:for_core_protos_cc",
         "@com_google_absl//absl/container:flat_hash_map",
-<<<<<<< HEAD
         #"@com_google_absl//absl/strings:string_view",
         "@pybind11",
-=======
-        "@com_google_absl//absl/strings:string_view",
```
Comment (draganmladjenovic) [inline tensorflow/compiler/mlir/quantization/tensorflow/python/BUILD]: Hmm. I think we can now pick the upstream one here.

File: third_party/xla/xla/backends/profiler/gpu/rocm_collector.cc (hunk 1 above)
Comment (i-chaochen) [reply third_party/xla/xla/backends/profiler/gpu/rocm_collector.cc:225]: Yes, I realise is clang caused this error on TF this morning as well..

**Comment by i-chaochen** [comment]
retest Ubuntu-GPU-single please

File: tensorflow/compiler/mlir/quantization/tensorflow/python/BUILD (hunk 4 above)
Comment (mmakevic-amd) [reply tensorflow/compiler/mlir/quantization/tensorflow/python/BUILD]: Fixed

File: third_party/xla/xla/client/lib/BUILD [hunk 5]
```diff
@@ -551,10 +551,9 @@ cc_library(
 xla_test(
     name = "self_adjoint_eig_test",
     srcs = ["self_adjoint_eig_test.cc"],
-    tags = ["no_rocm",],
     real_hardware_only = True,
     shard_count = 5,
-    tags = ["optonly"],
+    tags = ["no_rocm", "optonly"],
```
Comment (i-chaochen) [inline third_party/xla/xla/client/lib/BUILD]: I'm wondering is it possible to only skip that failed subtest instead of the whole test?

**Comment by i-chaochen** [comment]
retest Ubuntu-GPU-single please

**Comment by i-chaochen** [comment]
```
[2024-04-04T16:01:15.703Z] ERROR: Traceback (most recent call last):
[2024-04-04T16:01:15.703Z] 	File "/root/.cache/bazel/_bazel_root/fbac33eb30dbfb6b11b15a7ff5ac830d/external/local_xla/xla/mlir_hlo/tests/BUILD", line 11, column 13, in <toplevel>
[2024-04-04T16:01:15.703Z] 		lit_test(
[2024-04-04T16:01:15.703Z] 	File "/root/.cache/bazel/_bazel_root/fbac33eb30dbfb6b11b15a7ff5ac830d/external/llvm-project/llvm/lit_test.bzl", line 31, column 19, in lit_test
[2024-04-04T16:01:15.703Z] 		native.py_test(
[2024-04-04T16:01:15.703Z] Error in py_test: py_test(...) got multiple values for parameter 'deps'
[2024-04-04T16:01:15.703Z] ERROR: package contains errors: xla/mlir_hlo/tests
[2024-04-04T16:01:15.703Z] ERROR: package contains errors: xla/mlir_hlo/tests: Traceback (most recent call last):
[2024-04-04T16:01:15.703Z] 	File "/root/.cache/bazel/_bazel_root/fbac33eb30dbfb6b11b15a7ff5ac830d/external/local_xla/xla/mlir_hlo/tests/BUILD", line 11, column 13, in <toplevel>
[2024-04-04T16:01:15.703Z] 		lit_test(
[2024-04-04T16:01:15.703Z] 	File "/root/.cache/bazel/_bazel_root/fbac33eb30dbfb6b11b15a7ff5ac830d/external/llvm-project/llvm/lit_test.bzl", line 31, column 19, in lit_test
[2024-04-04T16:01:15.703Z] 		native.py_test(
[2024-04-04T16:01:15.703Z] Error in py_test: py_test(...) got multiple values for parameter 'deps'
[2024-04-04T16:01:15.703Z] WARNING: Target pattern parsing failed.
```

seems something wrong on xla/mlir_hlo/tests/BUILD", line 11  , could you reproduce it in your local?

**Review by i-chaochen — COMMENTED** [review]
This self_adjoint_eig_test.cc will be fixed in https://github.com/ROCm/frameworks-internal/issues/7682

File: third_party/xla/xla/service/gpu/conv_algorithm_picker.cc (hunk 2 above)
Comment (mmakevic-amd) [reply third_party/xla/xla/service/gpu/conv_algorithm_picker.cc:74]: Upstreamed, PR is [#11307](https://github.com/openxla/xla/pull/11307)

File: third_party/xla/xla/service/gpu/gpu_layout_assignment.cc (hunk 3 above)
Comment (mmakevic-amd) [reply third_party/xla/xla/service/gpu/gpu_layout_assignment.cc:58]: Same as above, PR [#11307](https://github.com/openxla/xla/pull/11307)

File: third_party/xla/xla/client/lib/BUILD (hunk 5 above)
Comment (mmakevic-amd) [reply third_party/xla/xla/client/lib/BUILD]: Done.

**Comment by mmakevic-amd** [comment]
Temporarily removed local deps in lit tests [(ref)](https://github.com/ROCm/tensorflow-upstream/blob/3b26854154c63257197af785e58bf6fed5acea97/third_party/xla/xla/mlir_hlo/tests/BUILD#L29)
This should be reverted once the upstream includes these changes from llvm: [lit_test : check if there is already a deps key in kwargs ](https://github.com/llvm/llvm-project/commit/dd5797505ebc2dbfdd58927c4f0a11a1256696eb)

**Comment by i-chaochen** [comment]
retest Ubuntu-GPU-single please

**Comment by i-chaochen** [comment]
retest Ubuntu-GPU-single please
retest Ubuntu-GPU-multi please

**Comment by i-chaochen** [comment]
retest Ubuntu-GPU-single please

**Comment by i-chaochen** [comment]
retest Ubuntu-GPU-single please

**Review by i-chaochen — APPROVED** [review]

**Pull request merged**